federation-js/src/composition/DirectiveMetadata.ts

```diff
@@ -192,3 +179,23 @@ export class DirectiveMetadata {
     }
   }
 }
+
+function mergeDirectiveUsages(
+  destination: DirectiveUsages | undefined,
+  source: DirectiveUsages,
+): DirectiveUsages {
+  let directiveUsages = destination;
+  if (!directiveUsages || directiveUsages.size === 0) {
+    directiveUsages = source;
```

Comment:
Why not ditch `directiveUsages` entirely, do an early return here, and just use `destination` in the rest of the function instead of `directiveUsages`? Is there a typing issue? Because I think it would make it more clear that directiveUsages is destination.

Reply:
e6c6c12